src/declarativewebutils.cpp

```diff
@@ -342,14 +342,9 @@ DeclarativeWebUtils *DeclarativeWebUtils::instance()
     return gSingleton;
 }
 
-QString DeclarativeWebUtils::downloadDir() const
-{
-    return QStandardPaths::writableLocation(QStandardPaths::DownloadLocation);
-}
-
 QString DeclarativeWebUtils::picturesDir() const
```

Comment:
We should drop this as well. Only used in ```BrowserContextMenu``` in its ```getUniqueFileName```.

The ```getUniqueFileName``` is also one of the places that uses ```WebUtils.fileExists```, the other place is in webutils.cpp itself. That makes me wondering if we should have some kind of FileUtils for creating ```uniqueFileName(fileName)```. I do not recall by heart if/where we are doing file checks etc on QML side.

To get this removed, you could add ```BrowserPaths``` to the root context. Better approach, that would not add extra stuff to root context, would be if we'd have a QObject wrapper and register this as singleton. ```BrowserPaths``` cannot be directly registered as singleton as qmlengine will take the ownership of it.